File: myoro_matchup_api_old/src/main/java/com/myoro/myoro_matchup_api/util/SubscriptionConstants.java
package com.myoro.myoro_matchup_api.util;

/** Constants for subscription functionality. */
public class SubscriptionConstants {
  /** Maximum number of games a free user can participate in. */
  public static final int FREE_USER_MAX_GAMES = 2;

  /** Monthly subscription price in USD. */
  public static final double MONTHLY_PRICE_USD = 7.00;

  /** iOS monthly product ID. */
  public static final String IOS_MONTHLY_PRODUCT_ID = "com.myoro.myoro_matchup.premium_monthly";

  /** Android monthly product ID. */
  public static final String ANDROID_MONTHLY_PRODUCT_ID = "premium_monthly";

  /** Private constructor to prevent instantiation. */
  private SubscriptionConstants() {}
}

File: myoro_matchup/lib/module/subscription/constant/subscription_constants.dart
/// Constants for subscription functionality.
final class SubscriptionConstants {
  SubscriptionConstants._();

  /// Monthly subscription price in USD.
  static const double monthlyPriceUsd = 7.00;

  /// Product ID for monthly subscription on iOS.
  /// TODO: Set up in App Store Connect.
  static const String iosMonthlyProductId = 'com.myoro.myoromatchup.premium_monthly';

  /// Product ID for monthly subscription on Android.
  /// TODO: Set up in Google Play Console.
  static const String androidMonthlyProductId = 'premium_monthly';

  /// Maximum number of games a free user can participate in.
  static const int freeUserMaxGameParticipation = 2;

  /// Legal URLs.
  /// TODO: Host these pages at antonkoetzler.github.io/legal
  static const String privacyPolicyUrl = 'https://antonkoetzler.github.io/legal/privacy-policy';
  static const String termsOfServiceUrl = 'https://antonkoetzler.github.io/legal/terms-of-service';
}
